File: Node/sendMessageMediaFromURL.js
const axios = require('axios'); 

// Substitua pelas suas credenciais
const api_key = 'GLOBAL_API_KEY';
const session_id = 'SESSION_ID';
/*PRESTE ATENÇÃO PARA NÃO ERRAR, O FORMATO DO NÚMERO É:
  CÓDIGO POSTAL DO PAÍS + DD + NÚMERO SEM O "9" NA FRENTE.
  ISSO MESMO SEM O "9" NA FRENTE POIS, O WHATSAPP TRANSFORMA
  O NÚMERO EM 8 DÍGITOS E NÃO 9 COMO É O NOVO PADRÃO BRASILEIRO.
  LOGO POR FAVOR UTILIZE OS OITO DÍGITOS E NÃO NOVE DÍGITOS
  COM O "9" NA FRENTE!!!

*/
const phone_number = '557183196364';

// Crie os dados da requisição
const data = {
    "chatId": "557183196364@c.us",
    "contentType": "MessageMediaFromURL",
    "content": "https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=Example"
  };

// Crie os cabeçalhos da requisição
const headers = {
  'accept': '*/*',
  'x-api-key': api_key,
  'Content-Type': 'application/json',
};

// Faça a requisição POST
axios.post(`https://zip-zop.site/client/sendMessage/${session_id}`, data, { headers })
  .then(response => {
    // Imprima a resposta
    console.log(response.data);
  })
  .catch(error => {
    console.error(error);
  });

File: Java/SendMessageMediaFromURL.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class SendMessageMediaFromURL {

    public static void main(String[] args) {
        // Substitua pelos valores reais
        String apiKey = "GLOBAL_API_KEY";
        String sessionId = "SESSION_ID";
        String phoneNumber = "557183196364";
        String urlString = "https://zip-zop.site/client/sendMessage/" + sessionId;
        String content = "https://api.qrserver.com/v1/create-qr-code/?size=150x150&data=Example";

        // Dados da mensagem
        String chatId = phoneNumber + "@c.us";

        HttpURLConnection connection = null;

        try {
            URL url = new URL(urlString);
            connection = (HttpURLConnection) url.openConnection();
            connection.setRequestMethod("POST");
            connection.setDoOutput(true);
            connection.setRequestProperty("Accept", "*/*");
            connection.setRequestProperty("x-api-key", apiKey);
            connection.setRequestProperty("Content-Type", "application/json");

            // Crie o corpo da requisição JSON
            String jsonInputString = String.format(
                "{\"chatId\": \"%s\", \"contentType\": \"MessageMediaFromURL\", \"content\": {\"%s\"}",
                chatId, content
            );

            // Envia os dados da requisição
            try (OutputStream os = connection.getOutputStream()) {
                byte[] input = jsonInputString.getBytes(StandardCharsets.UTF_8);
                os.write(input, 0, input.length);
            }

            // Obtém o código de resposta HTTP
            int responseCode = connection.getResponseCode();
            System.out.println("Response Code: " + responseCode);

            // Lê e exibe a resposta da requisição
            try (BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
                StringBuilder response = new StringBuilder();
                String line;
                while ((line = br.readLine()) != null) {
                    response.append(line).append("\n");
                }
                System.out.println("Response Body: " + response.toString());
            }

        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (connection != null) {
                connection.disconnect();
            }
        }
    }
}
